writer.go

```diff
@@ -180,3 +181,150 @@ func (w *FilteredLevelWriter) WriteLevel(level Level, p []byte) (int, error) {
 	}
 	return len(p), nil
 }
+
+var triggerWriterPool = &sync.Pool{
+	New: func() interface{} {
+		return new(bytes.Buffer)
+	},
+}
+
+// NewTriggerLevelWriter returns a new TriggerLevelWriter.
+//
+// It obtains a buffer from the pool and you must
+// call Close to return the buffer to the pool.
+func NewTriggerLevelWriter(w io.Writer, conditionalLevel, triggerLevel Level) *TriggerLevelWriter {
+	return &TriggerLevelWriter{
+		Writer:           w,
+		ConditionalLevel: conditionalLevel,
+		TriggerLevel:     triggerLevel,
+		buf:              triggerWriterPool.Get().(*bytes.Buffer),
```

Comment:
Extra optimization could be to get from the pool on first write instead of unconditionally.